test/modules/generic-mqtt-tester/src/lib.rs

```diff
@@ -91,3 +92,7 @@ pub enum MessageTesterError {
     #[error("received rejected subscription: {0}")]
     RejectedSubscription(String),
 }
+
+pub fn parse_sequence_number(publication: &ReceivedPublication) -> u32 {
+    publication.payload.slice(0..4).get_u32()
```

Comment:
Does this parse if an int is in the first 5 characters?

Reply:
The payload is structured as a u32 then potentially a bunch of garbage data to simulate large sized messages. This gets serialized to bytes. The deserialization logic needs to get the first 4 bytes (size of u32) to get the u32 out of the serialized value.